### ISSUE TITLE
Document @withRequest XML-RPC feature in the XML-RPC howto

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4732]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#4732|
|Type|enhancement|
|Created|2010-11-08 16:19:21Z|
|Branch|https://github.com/twisted/twisted/tree/withrequest-docs-4732|

Now that we finally have a way to pass request to xmlrpc methods, the documentation should demonstrate it so people know it's available.

<details><summary>Searchable metadata</summary>

```
trac-id__4732 4732
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__ 
branch__branches_withrequest_docs_4732 branches/withrequest-docs-4732
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1289233161000000 1289233161000000
changetime__1300659844000000 1300659844000000
version__None None
owner__exarkun exarkun
cc__wam cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [31247]) Branching to 'withrequest-docs-4732'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [31248]) Add an example of the withRequest decorator

refs [#4732](https://github.com/twisted/twisted/issues/4732)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set owner to @exarkun|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [31301]) Merge withrequest-docs-4732

Author: exarkun
Reviewer: jesstess
Fixes: [#4732](https://github.com/twisted/twisted/issues/4732)

Add documentation for the `withRequest` decorator which allows XML-RPC server
methods to receive the HTTP request object.
